bpf/lib/encrypt.h

```diff
@@ -205,6 +206,91 @@ do_decrypt(struct __ctx_buff *ctx, __u16 proto)
 #endif /* ENABLE_ROUTING */
 }
 
+static __always_inline int
+ipsec_maybe_redirect_to_encrypt(struct __ctx_buff *ctx, __be16 proto)
+{
+	struct remote_endpoint_info *dst = NULL;
+	struct remote_endpoint_info __maybe_unused *src = NULL;
+	void *data __maybe_unused, *data_end __maybe_unused;
+	struct iphdr __maybe_unused *ip4;
+	__u32 magic __maybe_unused = 0;
+	int ret = 0;
+	__u8 dst_mac = 0;
+
+	if (!eth_is_supported_ethertype(proto))
+		return DROP_UNSUPPORTED_L2;
+
```

Comment:
We're already receiving the `proto` as parameter :).

Reply:
Hm, this little bit of logic was ripped right from `wireguard.h`. Is it incorrect there as well?